Gm fix connection timeout

GabrieleManduchi opened the pull request:
Fix timeout parameter in GetAnswerInfoTS(). The argument was added but not reported outside mdstcpip. Therefore a random value was passed as timeout in java and C++ Connection objects. This is likely the cause for random failures in Connection tests during build.

A timeout in server side streaming management (used by jScope MdsStreamingDataProvider) has been also added to  avoid ghost mdsip processes hanging on a condition that will never be set.

GabrieleManduchi committed:
FIX: Connection timeout

A timeout argument has recently been added to mdstcpip/GetAnswerInfoTS(). However this routine was also used in
Connection C++ and java objects. Since there is no header in mdstcpip, the routine signature was manually reported in those libraries, and therefore the timeout was not reported and a random value was passed as timeout. This is likely also the cause for the random failures in Connection tests during build.

GabrieleManduchi committed:
FIX: removed debug message

removed a forgotten debug message

GabrieleManduchi committed:
FIX: streaming timeout

Added a timeout when waiting new data for server side stream management in order to give a chance to xinetd mdsip processes to exit when the connection is no more active.

tfredian reviewed (changes requested):
There are many more calls to GetAnswerInfoTS which also need repaired. I will try to track them down and fix them and push another commit to this branch.

tfredian committed:
Fix: Add timeout argument to calls to GetAnswerInfoTS

A timeout argument was added to the GetAnswerInfoTS routine but most of the calls to
that routine were modified to include this argument. This provides a timeout of 0
for those calls which disables timeouts on the calls (the original behavior). We may
want to consider enabling timeouts in certain instance where appropriate later.

AndreaRigoni reviewed (approved):
Ok for me !
